api/coupons.go

```diff
@@ -15,33 +14,32 @@ import (
 func (a *API) lookupCoupon(ctx context.Context, w http.ResponseWriter, code string) (*models.Coupon, error) {
 	couponCache := gcontext.GetCoupons(ctx)
 	if couponCache == nil {
-		notFoundError(w, "No coupons available")
-		return nil, errors.New("No coupons available")
+		return nil, notFoundError("No coupons available")
 	}
 
 	coupon, err := couponCache.Lookup(code)
 	if err != nil {
 		switch v := err.(type) {
 		case coupons.CouponNotFound:
-			notFoundError(w, v.Error())
+			return nil, notFoundError(v.Error())
```

Comment:
Since we know it is a `coupons.CouponNotFound` error, it will print precisely that.